src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,24 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String writeFile() {
+        return "D | " + super.writeFile() + " | " + this.by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " +
```

Comment:
I like how you adhere to the coding standard that any line length should be no longer than 120 chars by practicing proper indentation, which improved readability.